### PR TITLE
✨Added the Auth Link

### PR DESCRIPTION
**Pull request by HofmannZ**
This PR introduces the brand new Auth Link!

### Breaking changes

N/A

#### Fixes / Enhancements

- Added Auth Link class.

#### Docs

- Updated the example to use the new AuthLink.

### CODE REVIEW

File: lib/src/link/auth/link_auth.dart [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+class AuthLink extends Link {
+  AuthLink({
+    this.getToken,
+  }) : super(
+          request: (Operation operation, [NextLink forward]) {
+            StreamController<FetchResult> controller;
+
+            Future<void> onListen() async {
+              try {
+                final String token = getToken();
```
Comment (michael-golfi) [inline lib/src/link/auth/link_auth.dart]: Just checking in - is there a way here to also support async token calls?

Particularly in my use case (maybe others as well), I need to support cases where the token gets expired and needs to be refreshed and that would require doing the token call asynchronously.

Comment (micimize) [reply lib/src/link/auth/link_auth.dart]: I think #178 should solve this, if not go ahead and open a new issue

**Pull request merged**